### ISSUE TITLE
Refactor classes like RestIntegTestTask that do extended configuration in the constructor.

### ISSUE THREAD

**Issue by alpar-t** [elastic/elasticsearch#47804]
These tasks make configuration hard to follow.

We would like to replace their use with a plugin that sets up the task and necessary configuration for these types of tests.

**Comment by breskeby** [comment]
I've tackled this one in https://github.com/elastic/elasticsearch/pull/60261

**Comment by mark-vieira** [comment]
As part of refactoring to support things like lazy task evaluation and the configuration cache we no longer perform logic in task constructors.
